_sources/Strings/StringMethods.rst

```diff
@@ -145,10 +146,151 @@ change the original.  You can also consult the `Python documentation for strings
+Such a string is called a *format string*.  Places where
+braces are embedded are replaced by the value of an expression
+taken from the parameter list for the ``format`` method. There are many
+variations on the syntax between the braces. In this case we use
+the syntax where the first (and only) location in the string with
+braces has a substitution made from the first (and only) parameter.
+
+In the code above, this new string is assigned to the identifier
+``greeting``, and then the string is printed. 
+
+The identifier
+``greeting`` was introduced to break the operations into a clearer
+sequence of steps. However, since the value of ``greeting`` is only
+referenced once, it can be eliminated with the more concise
+version:
+
+.. activecode:: ch08_methods4
+
+    person = input('Enter your name: ') 
+    print('Hello {}!'.format(person)) 
+
+There can be multiple substitutions, with data of any type.  
+Next we use floats.  Try original price $2.50  with a 7% discount:
+
+.. activecode:: ch08_methods5
+
+    origPrice = float(input('Enter the original price: $')) 
+    discount = float(input('Enter discount percentage: ')) 
+    newPrice = (1 - discount/100)*origPrice
+    calculation = '${} discounted by {}% is ${}.'.format(origPrice, discount, newPrice)
+    print(calculation)
+
+The parameters are inserted into the braces in order.
+
+If you used the data suggested, this result is not satisfying.  
+Prices should appear with exactly two places beyond the decimal point,
+but that is not the default way to display floats.
+
+Format strings can give further information inside the braces 
+showing how to specially format data.
+In particular floats can be shown with a specific number of decimal places.  
+For two decimal places, put ``:.2f`` inside the braces for the monetary values:
+
+.. activecode:: ch08_methods6
+
+    origPrice = float(input('Enter the original price: $')) 
+    discount = float(input('Enter discount percentage: ')) 
+    newPrice = (1 - discount/100)*origPrice
+    calculation = '${:.2f} discounted by {}% is ${:.2f}.'.format(origPrice, discount, newPrice)
+    print(calculation)
+
+The 2 in the format modifier can be replaced by another integer to round to that
+specified number of digits.
+
+This kind of format string depends directly on the order of the
+parameters to the format method. There are other approaches that we will
+skip here, explicitly numbering substitutions and taking substitutions from a dictionary.
+
+A technical point: Since braces have special meaning in a format
+string, there must be a special rule if you want braces to actually
+be included in the final *formatted* string. The rule is to double
+the braces: ``{{`` and ``}}``. For example mathematical set
+notation uses braces. The initial and final doubled
+braces in the format string below generate literal braces in the
+formatted string:
+
+.. activecode:: ch08_methods7
```

Comment:
I love the addition of the format section.  In running through the book with this example, I did notice there was a bug in skulpt that may be worth mentioning.  The example you use prints out `{{5, 9}}` in Skulpt.  Even more confusingly removing the first { and the last } give the result you want in Skulpt, but cause an error in CPython.